nimble/host/src/ble_hs.c

```diff
@@ -388,6 +388,13 @@ ble_hs_reset(void)
     /* Clear configured addresses. */
     ble_hs_id_reset();
 
+    /* Clear adverising and scanning states. */
+    if (conn_handle == BLE_HS_CONN_HANDLE_NONE) {
```

Comment:
we should just call `ble_gap_reset_state` with a `ble_hs_reset_reason`.  Imho `conn_handle` at this point is always `BLE_HS_CONN_HANDLE_NONE`.

BTW later I think that we can move following block
```
 while (1) {

        conn_handle = ble_hs_atomic_first_conn_handle();
        if (conn_handle == BLE_HS_CONN_HANDLE_NONE) {
            break;
        }

        ble_gap_conn_broken(conn_handle, ble_hs_reset_reason);
    }
```
to the new `ble_gap_reset_state()`

Reply:
Makes sense :+1: